apps/moderation/serializers.py

```diff
@@ -1,22 +1,66 @@
 from django.core.serializers.python import Serializer
 from rest_framework import serializers
-from rest_framework.serializers import ModelSerializer
 
+common_instant_search_fields = {
+    'attributees': {'model': 'entities.entity'},
+    'related_entities': {'model': 'entities.entity'},
+    'tags': {'model': 'topics.topic'},
+    'location': {'model': 'places.place'},
+    'images': {'model': 'images.image'},
+}
 
-def get_moderated_model_serializer(
-    model_serializer: type[ModelSerializer],
-) -> type[ModelSerializer]:
-    """Return the serializer to be used for moderation of a moderated model.
-    Adds 'fields' field to given model_serializer.
-    """
 
-    class ModeratedModelSerializer(model_serializer):
-        fields = serializers.ReadOnlyField(source='get_moderated_fields')
+class ModeratedModelSerializer(serializers.ModelSerializer):
+    moderated_fields_excludes = ['id', 'meta', 'admin_url', 'absolute_url', 'cached_tags']
+
+    instant_search_fields = {}
+
+    def get_instant_search_fields(self) -> dict:
+        return common_instant_search_fields | self.instant_search_fields
```

Comment:
What is the rationale behind `common_instant_search_fields`? Could they not be defined on their corresponding serializers, or does it complicate inheritance?

Reply:
As the name suggests, these m2m fields are used in many modules. Extending serializers use instant_search_fields to define their own.